### PR TITLE
Unknown props: do not warn on React components.

### LINKED ISSUE #4049: it looks like an unknown prop is being sent through to the DOM, which will likely trigger a React console error.
Screen Shot 2023-06-22 at 6 22 45 PM

Hi everyone! Thanks for the incredible tool, guys! 
Could you please tell what to do if the prop is really needed for the underline react Button component?

### PR DESCRIPTION
**Pull request by woodreamz**
Edit the condition for the unknown props warning to only display it on HTML element and not React component.

Fixes #4049 #4071

### CODE REVIEW

**Comment by apolyanov** [comment]
Like @probablyup said we would more likely want to use the list of HTML tags that are in a Set here `packages/styled-components/src/utils/domElements.ts`. Apart from that, I happy with this change. Also if it's possible would you like to show me how you set up your local environment?

**Comment by woodreamz** [comment]
@probablyup I continue the conversation here to not "pollute" the issue.

I found a `isTag` function is used in the code, I decided to use it. Do you think it's better to check with the domElements list? I'm just wondering if there's a hit in performance.

**Comment by apolyanov** [comment]
@woodreamz also before you commit make sure to bump it to 6.0.4 as 6.0.3 was already released so you don't trigger more builds.

**Comment by probablyup** [comment]
> @woodreamz also before you commit make sure to bump it to 6.0.4 as 6.0.3 was already released so you don't trigger more builds.

Doesn't autorelease, I manually release the library at the moment so no worries there.

**Comment by apolyanov** [comment]
> > @woodreamz also before you commit make sure to bump it to 6.0.4 as 6.0.3 was already released so you don't trigger more builds.
> 
> Doesn't autorelease, I manually release the library at the moment so no worries there.

Just a habit of saving company money :D

**Comment by apolyanov** [comment]
@probablyup I am really sorry for the spam though, would you have time to help me set up styled-components locally. I am getting weird error when spinning up the playground app...

**Comment by woodreamz** [comment]
Ok, PR should be ready.

**Comment by woodreamz** [comment]
> @probablyup I am really sorry for the spam though, would you have time to help me set up styled-components locally. I am getting weird error when spinning up the playground app...

On my side, I just cloned the repo and did a `yarn install` as mentioned in[CONTRIBUTING.md](https://github.com/styled-components/styled-components/blob/main/CONTRIBUTING.md)

**Comment by apolyanov** [comment]
> > @probablyup I am really sorry for the spam though, would you have time to help me set up styled-components locally. I am getting weird error when spinning up the playground app...
> 
> On my side, I just cloned the repo and did a `yarn install` as mentioned in[CONTRIBUTING.md](https://github.com/styled-components/styled-components/blob/main/CONTRIBUTING.md)

What machine are you on?

**Comment by woodreamz** [comment]
> > > @probablyup I am really sorry for the spam though, would you have time to help me set up styled-components locally. I am getting weird error when spinning up the playground app...
> > 
> > 
> > On my side, I just cloned the repo and did a `yarn install` as mentioned in[CONTRIBUTING.md](https://github.com/styled-components/styled-components/blob/main/CONTRIBUTING.md)
> 
> What machine are you on? Can you give me a discord or something to message there?

I am running on macOS, I don't know what is not working on your side but I had some trouble running the sandbox because I did not build `styled-components`package locally so the link was not working. I could contact you on discord if I remember my account, let me check.

File: packages/styled-components/src/models/StyledComponent.ts [hunk 1]
```diff
@@ -151,7 +152,9 @@ function useStyledComponentImpl<Props extends object>(
         !shouldForwardProp &&
         process.env.NODE_ENV === 'development' &&
         !isPropValid(key) &&
-        !seenUnknownProps.has(key)
+        !seenUnknownProps.has(key) &&
+        // Only warn on DOM Element.
+        isDOMElement(elementToBeCreated)
```
[the remark this answers is not in the document]
Comment (woodreamz) [reply packages/styled-components/src/models/StyledComponent.ts]: No problem, I just remove isDOMElement and updated the condition.

**Review by probablyup — APPROVED** [review]
Thank you!

**Comment by AsuraKev** [comment]
Still getting issues with version 6.0.7

**Pull request merged**